### DISCUSSION TITLE
Show source of program compiled with Mingw-w64 gcc

### DISCUSSION THREAD

**Discussion by 3N4N** [x64dbg/x64dbg#2776] [Q&A] [answered]
I'm new to x64dbg. Didn't even know about it. So I wanted to check if it works with mingw-w64 compiled executable. It works, but doesn't show the source.

I'm using the basic hello world c program and compiling it with the Mingw-w64 `gcc` program with `-g` flag.

```bash
e:\projects\hello> C:\msys64\mingw64\bin\gcc.exe -g .\main.c -o prog.exe
```

I also checked with the Mingw-w64 `gdb` and it shows the source, so the `-g` flag must have worked.

And I checked the manual provided in the release package. The `chm` file. It doesn't say anything about code sources.

**Comment by mrexodia** [comment, marked as answer]
In principle x64dbg is not a source-level debugger so you’ll probably have a bad time.

You can use [cv2pdb](https://github.com/rainers/cv2pdb) to convert the DWARF format to PDB to see the line numbers and open the sources.
